### PR TITLE
Helper changes

### PR DESCRIPTION
**Pull request by sharadregoti**
Don't forget to create a new release after merging this pull request.

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -74,16 +72,41 @@ func getLogLevel(logLevel string) zapcore.Level {
...
+	message  string
+	rawError string
+}
+
+func (e helperError) Message() string {
+	return e.message
+}
+
+func (e helperError) Error() string {
+	return e.rawError
```
Comment (YourTechBud) [inline logger.go]: I suggest you return the message here. Make a RawError function to return the raw error. What say?

File: logger.go [hunk 2]
```diff
@@ -74,16 +72,41 @@ func getLogLevel(logLevel string) zapcore.Level {
...
+}
+
 // LogError logs the error in the proper format
 func (l *logger) LogError(requestID, message string, err error, fields map[string]interface{}) error {
-	// Log the error
-	if fields != nil {
-		zapLogger.Error(message, zap.Any("error", err), zap.String("requestId", requestID), zap.Any("fields", fields))
-	} else {
-		zapLogger.Error(message, zap.Any("error", err), zap.String("requestId", requestID))
+	switch temp := err.(type) {
```
Comment (YourTechBud) [inline logger.go]: Instead of using type switch, simply do a type assert. If ok then return as is.

File: logger.go [hunk 3]
```diff
@@ -74,16 +72,41 @@ func getLogLevel(logLevel string) zapcore.Level {
 	}
 }
 
+type Error interface {
```
Comment (YourTechBud) [inline logger.go]: We don't need to make this an interface. Let this be a struct

File: logger.go [hunk 4]
```diff
@@ -74,16 +72,41 @@ func getLogLevel(logLevel string) zapcore.Level {
...
-	} else {
-		zapLogger.Error(message, zap.Any("error", err), zap.String("requestId", requestID))
+	switch temp := err.(type) {
+	case Error:
+		return temp
+	case error:
+		var v Error = helperError{message: message, rawError: err.Error()}
+		// Log the error
+		if fields != nil {
+			zapLogger.Error(message, zap.Any("error", v.Error()), zap.String("requestId", requestID), zap.Any("fields", fields))
```
Comment (YourTechBud) [inline logger.go]:
```suggestion
			zapLogger.Error(message, zap.Any("error", err.Error()), zap.String("requestId", requestID), zap.Any("fields", fields))
```

File: logger.go [hunk 5]
```diff
@@ -74,16 +72,41 @@ func getLogLevel(logLevel string) zapcore.Level {
...
+		} else {
+			zapLogger.Error(message, zap.Any("error", v.Error()), zap.String("requestId", requestID))
```
Comment (YourTechBud) [inline logger.go]:
```suggestion
			zapLogger.Error(message, zap.Any("error", err.Error()), zap.String("requestId", requestID))
```

**Review by YourTechBud — CHANGES_REQUESTED** [review]

File: logger.go (hunk 1 above)
Comment (sharadregoti) [reply logger.go]: Yes

File: logger.go [hunk 6]
```diff
@@ -74,16 +74,40 @@ func getLogLevel(logLevel string) zapcore.Level {
 	}
 }
 
+// type Error interface {
```
Comment (YourTechBud) [inline logger.go]: Remove this

**Review by YourTechBud — CHANGES_REQUESTED** [review]

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**